### DISCUSSION TITLE
Extremely slow speed on AMD cards

### DISCUSSION THREAD

**Discussion by Shizzins** [lshqqytiger/stable-diffusion-webui-directml#50] [General]
The WebUI's speed for me is 4 s/it and it's extremely slow. SHARK by nod AI is very fast for AMD users, but it's very limited in terms of features. Direct ML is extremely slow, and I think torch-mlir would be amazing to implement. Please see to it, us AMD users are very left out.

**Comment by Miraihi** [comment]
If you think that DirectML is slow then you haven't seen the onyx iteration. It's a massive improvement and I hope that it gets even better with future directml updates.

**Reply by lshqqytiger** [reply]
In fact, DirectML couldn't even run webui until a few months ago. It is improving, but it has yet to show significant progress in memory management.

**Reply by Omniphantasm** [reply]
Man, I dunno what the fuck I did but this was running fine a 512x512 took 10-15 seconds tops for me and then I updated this thing, broke the shit out of my install and had to start from scratch and now its taking minutes to make a 512x512 and dragging along at a 9-10s/it

I have no idea what magic setting I did before to get this working well on my 7900 XTX and R9 7950X but I have definitely broke something.

**Reply by ponyguru** [reply]
Yeah, I updated my install today and it no longer runs at all. I am not sure exactly what broke.

**Reply by Yodrik** [reply]
Lately I have small adjustments to the point I should do every pull in a new folder. For the latest commits, I experienced #79 after the latest pull and removing "venv" folder, and allowing that to refresh itself fixed it.

**Comment by Lumpsack** [comment]
I found this version to be quicker than Shark on Windows with a 7900 XTX, that said memory on shark is better - but oddly results worse (which I wasn't expecting), so kudos to Ishqqytiger for doing this - without this I'd be £s worse off having to go for the nvidia.
